### PR TITLE
updated ozone adapter from v1.4.4 -> v2.0.0

### PR DESCRIPTION
**Pull request by afsheenb**
- [X] Feature
- [X] Refactoring

## Description of change
This updated includes several changes:

* Creates an adId key-value for DFP ad selection for each SSP (see https://github.com/prebid/prebid-server/issues/860 for background)
* Creates a dealID DFP KVP for each SSP should response contain a dealId field
* Utilises PriceGranularity settings on page to provide pricing DFP KVP keys for rounded pricing
* Creates a DFP KVP for the Ozone Adapter Version
* Initial Outstream Video Support

- contact email of the adapter’s maintainer
engineering@ozoneproject.com

### CODE REVIEW

**Comment by jsnellbaker** [comment]
Hi @afsheenb I see you've been making some updates (thanks!), but it seems like there are some lint errors present with the last commit.  Can you take a look at the circleci test results (click the Details link in the section below) and the few changes?  

You could also just run `gulp lint` in your project to have the issues be auto-fixed and then just add/commit those fixes into the PR.

**Comment by AskRupert-DM** [comment]
hey @jsnellbaker - apologies for the multiple submissions but think this is now good to review.

- you should see an updated MD file, your previous feedback around some of the code updated in the bidadapter JS, and some sample configs for testing outstream included within the MD file.

Regarding initial support for outstream and your earlier question: currently we only respond with outstream videos from Unruly, and they have requested that we render their creatives with their renderer. In the future, we will deliver other outstream videos and will utilize our own renderer, but this will only be a change in our bidder's response and not require changes client side. 

Also should flag some caveats with being able to test outstream:

(a) right now we seem to only be getting/delivering valid responses via UK IPs - working with the partner to provide a non-geo targeted response but you may need to improvise here.

(b) if you are testing using your own testpage - you will need to either spoof the request so its coming from one of our whitelisted domains (silvermine.io can be used) 

I will try provide an update here with a link to a testpage that allows you to easily overcome the caveats 
I mention above once I've been able to get a non-geo targeted response setup for outstream. The banners should however work for you correctly.

File: modules/ozoneBidAdapter.js [hunk 1]
```diff
@@ -90,76 +92,90 @@ export const spec = {
...
-      if (ozoneBidRequest.hasOwnProperty('crumbs') && ozoneBidRequest.crumbs.hasOwnProperty('pubcid')) {
-        obj.ext.ozone.pubcid = ozoneBidRequest.crumbs.pubcid;
-      }
-      return obj;
-    });
-    utils.logInfo('tosendtags = ', tosendtags);
+        if (ozoneBidRequest.params.hasOwnProperty('lotameData')) {
+          obj.ext.ozone.lotameData = ozoneBidRequest.params.lotameData;
+        }
+        if (ozoneBidRequest.hasOwnProperty('crumbs') && ozoneBidRequest.crumbs.hasOwnProperty('pubcid')) {
```
[the remark this answers is not in the document]
Comment (AskRupert-DM) [reply modules/ozoneBidAdapter.js]: updated - thanks.

File: modules/ozoneBidAdapter.js [hunk 2]
```diff
@@ -90,76 +92,90 @@ export const spec = {
...
+              utils.logInfo('OZONE: [WARNING] Only outstream video is supported. Rejecting bid: ', ozoneBidRequest);
+              return false;
+            }
+          }
+          return true;
+        })
+      .map(ozoneBidRequest => {
+        var obj = {};
+        obj.id = ozoneBidRequest.bidId; // this causes an error if we change it to something else, even if you update the bidRequest object: "WARNING: Bidder ozone made bid for unknown request ID: mb7953.859498327448. Ignoring."
+        // obj.id = ozoneBidRequest.adUnitCode; // (eg. 'mpu' or 'leaderboard') A unique identifier for this impression within the context of the bid request (typically, starts with 1 and increments.
```
[the remark this answers is not in the document]
Comment (AskRupert-DM) [reply modules/ozoneBidAdapter.js]: removed - thanks

**Comment by jsnellbaker** [comment]
Thanks for the updates, clarifications and test pages @afsheenb. 
The changes look good now.

**Pull request merged**